### ISSUE TITLE
Difference of Time with format 'datetime' returns TimeDelta with format 'jd'

### ISSUE THREAD

**Issue by Holt59** [astropy/astropy#15297]
### Description

When computing the difference between two `Time` objects with format `datetime`, the returned `TimeDelta` as format `jd`, which creates accuracy errors in computation.

### Expected behavior

When computing the difference of two `Time` with format `datetime`, the resulting `TimeDelta` objects should have `datetime` format and be exact.

### How to Reproduce

```python
Time(datetime(2023, 7, 1, 8)) - Time(datetime(2023, 7, 1, 6))
# outputs: <TimeDelta object: scale='tai' format='jd' value=0.08333333333333331>
# expected: <TimeDelta object: scale='tai' format='datetime' value=2:00:00>
```

### Versions

```
Windows-10-10.0.19045-SP0
Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)]
astropy 5.3.2
Numpy 1.25.2
pyerfa 2.0.0.3
Scipy 1.11.2
Matplotlib 3.7.2
```

**Comment by datajungler** [comment]
The subtraction method for Time object is to return a TimeDelta object with 'jd' format.

A quick solution to ensure the expected behavior is to replicate the TimeDelta object with 'datetime' format.
`td = Time(datetime(2023, 7, 1, 8)) - Time(datetime(2023, 7, 1, 6))`
`td = td.replicate(format='datetime')`


In the pull request, I attempted to incorporate additional lines of code to verify if the format of the time object is in 'datetime' format before returning a TimeDelta object in the same format. Hope this would help.

**Comment by taldcroft** [comment]
@Holt59 - thanks for reporting this issue, this is an interesting point of expected behavior.

A starting point is to understand that the `datetime` format output is not actually more accurate than the returned `TimeDelta` object with format `jd`. Internally the `TimeDelta` object has 128 bits of precision regardless of the format. So in your example you can do this:
```
>>> td = Time(datetime(2023, 7, 1, 8)) - Time(datetime(2023, 7, 1, 6))
>>> td.datetime
datetime.timedelta(seconds=7200)
```
The apparent "exact" result you see in `datetime.timedelta` is only because that class only represents times to within 1 microsec. So for example:
```
>>> td = Time(datetime(2023, 7, 1, 8)) - Time(datetime(2023, 7, 1, 6)) + 0.1 * u.us
>>> td.datetime
datetime.timedelta(seconds=7200)
>>> td.sec
7200.000000099998
```
`Time` and `TimeDelta` computations are done to similar accuracy as seen here (better than around 1e-5 microsec).

The takeaway is that in the original example it would be most accurate to use the `sec` format for a single 64-bit representation. The `datetime` format is less accurate.

**Comment by Holt59** [comment]
@taldcroft Thanks for the answer. 

I probably tried to narrow the issue too much, hiding the initial problem. For me the problem is that if I have two `Time` objects with format `datetime`,  I expect computation to be exact according to `datetime`, i.e., the return `TimeDelta` in my example should contains *exactly* 7200s (whatever the underlying format).

My original problem involves more computation, typically:

```python
from datetime import datetime
from astropy.time import Time

t1, t2, t3 = datetime(2023, 7, 1, 8), datetime(2023, 7, 1, 6), datetime(2023, 7, 1, 22)

# without astropy, the result is exactly 0.125 (can be stored exactly)
(t1 - t2) / (t3 - t2)

# with astropy, result is inexact
((Time(t1) - Time(t2)) / (Time(t3) - Time(t2))).value
```

I understand that the precision is pretty high, but I was not expecting a different between the two computations here.

**Comment by taldcroft** [comment]
@Holt59 - I can understand the confusion. However the fundamental implementation of astropy time uses a 128-bit floating point representation of time or delta time in Julian days. The *format* that you see with `.value` is only an output representation of the internal 128-bit time, and when you change the format there is no change in the internal 128-bit value. All computations are done with the internal 128-bit value to maintain the sub-nanosec precision regardless of format.

As with any floating point computation there will be values that cannot be represented *exactly*. Astropy Time provides ways to handle this fundamental issue, including the `isclose()` method. This is similar to what you need to do frequently in numpy float comparisons:
```
In [20]: t1 = Time(datetime(2020, 1, 1, 1))
    ...: t2 = Time(datetime(2020, 1, 1, 3))
    ...: dt = t2 - t1

In [21]: dt.isclose(TimeDelta(timedelta(hours=2)))
Out[21]: True
```

**Comment by taldcroft** [comment]
@Holt59 - would #15334 help? I'd like to close this issue with that PR.

**Comment by Holt59** [comment]
@taldcroft It would not fix the issue but would clarify the behavior for future users at least.

**Comment by taldcroft** [comment]
@Holt59 - OK. This will be *closed* with #15334 but the intent of your original issue won't be fixed since that would require changing some fundamentals of the sub-package and/or introducing undesired API changes.
